Set more precise React/React-Native package version dependencies for Example

abrahambotros opened the pull request:
- In general, try to avoid breaking changes caused by >=
  - In particular, such a breaking change can be seen by testing the current Example
    app on Android with the existing package.json's and RN 0.43+, which causes errors
    in MainApplication public/protected method settings.

rt2zz commented:
I agree with fixing the example app, but for peer deps things get trickier, especially when people want to use rc's.

abrahambotros commented:
@rt2zz, ah jeez of course, temporarily lapse of judgment there on my part, big thanks for correcting! Reverted changes to the main `package.json`, leaving for Example's `package.json`, and updated PR title accordingly.

rt2zz commented:
many thanks!
